app/ide-desktop/lib/dashboard/src/utilities/safeJsonParse.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @file
+ *
+ * A utility function to safely parse a JSON string.
+ * returns the default value if the JSON string is invalid.
+ * Also provides a type for the parsed JSON.
+ */
+
+/**
+ * Safely parse a JSON string.
+ * Parse the JSON string and return the default value if the JSON string is invalid.
+ * Or if the parsed JSON does not match the type assertion.
+ */
+export function safeJsonParse<T = unknown>(
+  value: string,
+  defaultValue: T,
+  assert: (parsed: unknown) => parsed is T
```

Comment:
not sure about the naming of `assert`

Reply:
Alternatives?

Reply:
> imo "assert" is a function that errors when its condition fails

If we look at the description(and the semantic) of the function:

```
/**
 * Safely parse a JSON string.
 * Parse the JSON string and return the default value if the JSON string is invalid.
 * Or if the parsed JSON does not match the type assertion.
 */
```

That means if the json is invalid we return `defautValue` and it is not that important(from the user of the function perspective) if assert function throws or not. Imo, assert here is good enough: It _asserts_ that the value matches the expected type.

Reply:
> It *asserts* that the value matches the expected type.

no it does not - it *returns a boolean* that tells the caller wheether or not the value matches the expected type.

`assert`s that error, follow not only TS' definition of `assert`, but also rust, c++, java, c# and lua's definitions (among other languages).

Reply:
iow: *asserting* something means saying "yes this is the case". but predicates may return `false`, so they are not assertions